vnext/JSI/Shared/ChakraApi.h

```diff
@@ -0,0 +1,468 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
+#define USE_EDGEMODE_JSRT
+#endif
+#include <jsrt.h>
+#endif
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <string_view>
+
+// We use macros to report errors.
+// Macros provide more flexibility to show assert and provide failure context.
+
+// Check condition and crash process if it fails.
+#define ChakraVerifyElseCrash(condition, message)          \
+  do {                                                     \
+    if (!(condition)) {                                    \
+      assert(false && "Failed: " #condition && (message)); \
+      *((int *)0) = 1;                                     \
+    }                                                      \
+  } while (false)
+
+// Throw native exception
+#define ChakraThrow(message) ThrowNativeException(message);
+
+// Check condition and throw native exception if it fails.
+#define ChakraVerifyElseThrow(condition, message) \
+  do {                                            \
+    if (!(condition)) {                           \
+      ThrowNativeException(message);              \
+    }                                             \
+  } while (false)
+
+// Evaluate expression and throw JS exception if it fails.
+#define ChakraVerifyJsErrorElseThrow(expression) \
+  do {                                           \
+    JsErrorCode temp_error_code_ = (expression); \
+    if (temp_error_code_ != JsNoError) {         \
+      ThrowJsException(temp_error_code_);        \
+    }                                            \
+  } while (false)
+
+namespace Microsoft::JSI {
+
+/**
+ * @brief A wrapper for Chakra APIs.
+ *
+ * The ChakraApi class wraps up the Chakra API functions in a way that:
+ * - functions throw exceptions instead of returning error code (derived class can define the exception types);
+ * - standard library types are used when possible to simplify usage.
+ *
+ * The order of the functions below follows the corresponding functions order in the chakrart.h
+ * Currently we only wrap up functions that are needed to implement the JSI API.
+ */
+struct ChakraApi {
+  /**
```

Comment:
final

Reply:
I am not sure what your comment is saying. If it is about declaring this file final, then no, we inherit the ChakraRuntime from this class.

---
In reply to: [470459946](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470459946) [](ancestors = 470459946)